src/Repository.js

```diff
@@ -43,7 +44,34 @@ export default class Repository {
     return this.lernaJson && this.lernaJson.bootstrapConfig || {};
   }
 
+  get packages() {
+    if (!this._packages) {
+      this.buildPackageGraph();
+    }
+    return this._packages;
+  }
+
+  get filteredPackages() {
+    if (!this._filteredPackages) {
+      this.buildPackageGraph();
+    }
+    return this._filteredPackages;
+  }
+
+  get packageGraph() {
+    if (!this._packageGraph) {
+      this.buildPackageGraph();
```

Comment:
Hmmm.. I liked the idea of pulling the filtering out of the commands just to make them consistent. Maybe these getters should each take optional { scope, ignore } flags?